src/core/annotation.js

```diff
@@ -1985,11 +1985,11 @@ class WidgetAnnotation extends Annotation {
     if (!this.data.combo) {
       value = value.trim();
     } else {
-      // The value is supposed to be one of the exportValue.
-      const option =
-        this.data.options.find(({ exportValue }) => value === exportValue) ||
-        this.data.options[0];
-      value = (option && option.displayValue) || "";
+      // The value can be one of the exportValue or any other values.
+      const option = this.data.options.find(
+        ({ exportValue }) => value === exportValue
+      );
+      value = (option && option.displayValue) || value;
```

Comment:
Should we perhaps trim any spaces in this case as well?

GitHub won't let me add a suggestion on code outside of the diff, but basically [change this code ](https://github.com/mozilla/pdf.js/blob/972744a68f8e3d0aea03225c7142693da76d0569/src/core/annotation.js#L1983-L1993) into something like:
```javascript
assert(typeof value === "string", "Expected `value` to be a string.");
value = value.trim();

if (this.data.combo) {
  // The value can be one of the exportValue or any other values.
  const option = this.data.options.find(
    ({ exportValue }) => value === exportValue
  );
  value = (option && option.displayValue) || value;
}
```